### ISSUE TITLE
Allow unicode in email local part

### ISSUE THREAD

**Issue by mastre** [sendgrid/sendgrid-php#920]
#### Issue Summary

Sending email to an email address which has unicode characters in local part (example: `françois@domain.tld`) results in an error using the `sendgrid-php` library.

#### Steps to Reproduce
```
...
$email = new \SendGrid\Mail\Mail();
$email->setSubject('My Subject');
$email->addTo('françois@domain.tld');
...
```
results in
```
PHP Fatal error:  Uncaught SendGrid\Mail\TypeException: $emailAddress must be valid and of type string. in ...
#0 .../vendor/sendgrid/sendgrid/lib/mail/EmailAddress.php(51): SendGrid\Mail\EmailAddress->setEmailAddress('fran\xC3\xA7ois...')
...
```
This is triggered by
https://github.com/sendgrid/sendgrid-php/blob/f64b6fda6d6253e1b2e1bd630764bead31009496/lib/mail/EmailAddress.php#L74
PHP allows an additional flag `FILTER_FLAG_EMAIL_UNICODE` to `filter_var()` when used with `FILTER_VALIDATE_EMAIL` to allow unicode in the local part of an email address (the part before the `@`).  Unless there is a particularly good reason to block this here, a simple change is needed on the above referenced line:
```filter_var($emailAddress, FILTER_VALIDATE_EMAIL, FILTER_FLAG_EMAIL_UNICODE)```
I can create a pull request.

#### Technical details:

* sendgrid-php Version: master 7.3.0 (latest commit: 37fa19d3ae019842f07a2a43e92ed0f566ad927d)
* PHP Version: 7.2

**Comment by childish-sambino** [comment]
Happy to review a PR if you'd like to submit one.

**Comment by kampalex** [comment]
Flag `FILTER_FLAG_EMAIL_UNICODE` is available from PHP 7.1, but this library supports PHP 5.6 and newer.
I suggest implementing this:
```php
[...]
//  Define additional flags for filter_var to verify unicode characters on local part
//  Constrant FILTER_FLAG_EMAIL_UNICODE is available since PHP 7.1
$flags = (defined('FILTER_FLAG_EMAIL_UNICODE')) ? FILTER_FLAG_EMAIL_UNICODE : null;
if (!(is_string($emailAddress) && filter_var($emailAddress, FILTER_VALIDATE_EMAIL, $flags))) {
[...]
```
Because it's used in 3 files, maybe a static method `isValidAddress` (responding boolean) in class `EmailAddress` is a solution?
